src/mono/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.xml

```diff
@@ -177,4 +177,57 @@
       <attribute internal="RemoveAttributeInstances" />
     </type>
   </assembly>
+
+  <!-- Metadata attributes present in most core SDKs assemblies -->
+  <assembly fullname="*">
+    <type fullname="System.Reflection.AssemblyMetadataAttribute">
+      <attribute internal="RemoveAttributeInstances">
+        <argument type="System.Object">
+          <argument>Serviceable</argument>
+        </argument>
+      </attribute>
+    </type>
+    <type fullname="System.Reflection.AssemblyMetadataAttribute">
+      <attribute internal="RemoveAttributeInstances">
+        <argument type="System.Object">
+          <argument>PreferInbox</argument>
+        </argument>
+      </attribute>
+    </type>
+    <type fullname="System.Reflection.AssemblyMetadataAttribute">
+      <attribute internal="RemoveAttributeInstances">
+        <argument type="System.Object">
+          <argument>.NETFrameworkAssembly</argument>
+        </argument>
+      </attribute>
+    </type>
+    <type fullname="System.Reflection.AssemblyMetadataAttribute">
+      <attribute internal="RemoveAttributeInstances">
+        <argument type="System.Object">
+          <argument>RepositoryUrl</argument>
+        </argument>
+      </attribute>
+    </type>
+    <type fullname="System.Reflection.AssemblyMetadataAttribute">
+      <attribute internal="RemoveAttributeInstances">
+        <argument type="System.Object">
+          <argument>SourceCommitUrl</argument>
+        </argument>
+      </attribute>
+    </type>
+    <type fullname="System.Reflection.AssemblyMetadataAttribute">
+      <attribute internal="RemoveAttributeInstances">
+        <argument type="System.Object">
+          <argument>CommitHash</argument>
```

Comment:
This is pretty useful when trying to diagnose issues. Are we sure we want to remove this?

Reply:
I guess we already have:

```
[assembly: AssemblyInformationalVersion("6.0.0-preview.4.21181.1+8cc37cb9d8405be0b15bdc8ae9cf2ddc31dbd998")]
```

So as long as we aren't removing that, we should be OK.

Reply:
Yep, AssemblyInformationalVersion stays and can be used for such purposed.

Reply:
I see only ASP.NET has CommitHash and SourceCommitUrl

https://github.com/dotnet/aspnetcore/blob/340d161093d01e8c7c9ffea4e4dbb50838d893e9/eng/targets/CSharp.Common.targets#L51-L59

These seem to be duplicate information of RepositoryUrl and AssemblyInformationalVersion, which come from the official SDK.

https://github.com/dotnet/sdk/blob/446589256a6fc128d1adffe83551a727ebf7f234/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets#L60-L67
https://github.com/dotnet/sdk/blob/446589256a6fc128d1adffe83551a727ebf7f234/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets#L100-L104

I wonder if we can just remove them from ASP.NET's build entirely? That way we don't need to remove them here.

cc @pranavkm @captainsafia - thoughts?

Reply:
My intuition is that we should remove them from aspnetcore's build if we're doing something we don't need to be doing here.

@dougbu Insight on why we currently have our own logic for generating this assembly info instead of using what's in the SDK?

Reply:
I've opened https://github.com/dotnet/aspnetcore/issues/31621 to track this.